api/src/main/java/javax/enterprise/event/TransactionPhase.java

```diff
@@ -26,10 +26,14 @@
  * Transactional observer methods are observer methods which receive event notifications during the before or after completion
  * phase of the transaction in which the event was fired. If no transaction is in progress when the event is fired, they are
  * notified at the same time as other observers.
+ * If the transaction is in an illegal state or already rolled back when the event is fired, the {@link #BEFORE_COMPLETION},
```

Comment:
@struberg Could you please specify what exactly is an 'illegal state' here? I checked the spec, but still can't pinpoint that.